solr/solrj/src/java/org/apache/solr/common/cloud/DocCollection.java

```diff
@@ -192,6 +193,28 @@ public DocRouter getRouter() {
   }
 
   @Override
+  public boolean equals(Object o) {
+    if (o == this) {
+      return true;
+    }
+    if (o.getClass() != this.getClass()) {
+      return false;
+    }
+    DocCollection other = (DocCollection) o;
+    return super.equals(other) // check equality of inherited properties
+        && znodeVersion == other.znodeVersion
+        && name.equals(other.name)
+        && slices.equals(other.slices)
+        && znode.equals(other.znode)
+        && Objects.equals(router, other.router);
+  }
+
+  @Override
+  public int hashCode() {
+    return super.hashCode() ^ Objects.hash(znodeVersion, name, slices, router, znode);
+  }
+
+  @Override
```

Comment:
this (and the others) seem kind of invasive for a monkey patch.. maybe just compare the string output in your test?